Sources/Sentry/SentryTime.mm

```diff
@@ -10,9 +10,15 @@
 timeIntervalToNanoseconds(double seconds)
 {
     NSCAssert(seconds >= 0, @"Seconds must be a positive value");
-    NSCAssert(seconds <= UINT64_MAX / 1e9,
+    NSCAssert(seconds <= (double)UINT64_MAX / (double)NSEC_PER_SEC,
         @"Value of seconds is too great; will overflow if casted to a uint64_t");
-    return (uint64_t)(seconds * 1e9);
+    return (uint64_t)(seconds * NSEC_PER_SEC);
+}
+
+double
```

Comment:
```suggestion
NSTimeInterval
```

Reply:
We can't use `NSTimeInterval` because of `SENTRY_EXTERN_C_BEGIN`. I think it's fine to keep the double. I'm not that familiar with the C and C++ interop. What would we need to do to remove the `SENTRY_EXTERN_C_BEGIN`, @armcknight?